html/pfappserver/root/static.alt/src/components/pfFormUpload.vue

```diff
@@ -70,67 +78,136 @@
     <slot>
       <b-button class="ml-3"><icon name="upload"></icon> {{ $t('Upload') }}</b-button>
     </slot>
+    <b-modal v-if="showErrorModal" v-model="showErrorModal" centered
+      :title="$t('Upload Error')"
+      @hide="clearFirstError()"
+    >
+      <b-media>
+        <icon name="exclamation-triangle" scale="2" slot="aside" class="text-danger"></icon>
+        <h4>{{ firstError.name }}</h4>
+        <p class="font-weight-light mt-3 mb-0">{{ firstError.error.message }}</p>
+        <p class="font-weight-light mt-3 mb-0 text-pre text-black-50">Ref: {{ firstError.error.name }} (#{{ firstError.error.code}})</p>
+      </b-media>
+      <div slot="modal-footer">
+        <b-button variant="primary" @click="clearFirstError()">{{ $t('Continue') }}</b-button>
+      </div>
+    </b-modal>
   </div>
 </template>
 
 <script>
+import FileStore from '@/store/base/file'
+
 export default {
   name: 'pf-form-upload',
   props: {
     accept: {
       type: String,
       default: '*/*'
     },
+    encoding: {
+      type: String,
+      default: 'utf-8'
+    },
     multiple: {
       type: Boolean,
       default: false
     },
-    files: {
-      type: Array,
-      default: () => []
-    },
     cumulative: {
       type: Boolean,
       default: false
     },
     title: {
       type: String,
       default: ''
+    },
+    readAsText: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      files: [],
+      showErrorModal: false
     }
   },
   methods: {
+    storeNameFromFile (file) {
+      const { lastModified, name } = file
+      return `file/${name}/${lastModified}`
+    },
     uploadFiles (event) {
       if (!this.cumulative) {
         this.files = []
       }
-      const files = event.target.files
+      const { target: { files } = {} } = event
       Array.from(files).forEach((file, index, files) => {
-        let reader = new FileReader()
-        reader.onload = ((localfile) => {
-          return (e) => {
-            let newfile = {
-              result: e.target.result,
-              lastModified: localfile.lastModified,
-              name: localfile.name,
-              size: localfile.size,
-              type: localfile.type
-            }
-            // prevent duplicates
-            if (this.files.map(file => { return file.name + file.lastModified }).includes(newfile.name + newfile.lastModified)) return
-            this.files.push(newfile)
-            if (this.cumulative || this.files.length === files.length) {
-              this.$emit('load', this.files)
-            }
+        const fileIndex = this.files.findIndex(f => f.name === file.name && f.lastModified === file.lastModified)
+        if (fileIndex > -1) {
+          this.$emit('focus', fileIndex)
```

Comment:
I think it would also be nice to focus on the newly created tab/pill when opening a new file.

Reply:
Resolved in https://github.com/inverse-inc/packetfence/pull/4827/commits/0d429ebae2b06e7be15e4fe9ea4022136bdfded7